src/components/appointment/Appointment.js

```diff
@@ -0,0 +1,134 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import swal from 'sweetalert';
+import './appointment.css';
+import { useParams, useNavigate } from 'react-router-dom';
+import { TimePicker, DatePicker } from '@mui/x-date-pickers';
+import { postData } from '../../redux/appointments/appointmentSlice';
+
+const Appointment = () => {
```

Comment:
- [ ] I can't reserve doctors:

![image](https://github.com/kamranasif506/Book-A-Doctor/assets/57421823/2aeba71e-723b-4603-8921-7fc9d417c092)

Server console output:

![image](https://github.com/kamranasif506/Book-A-Doctor/assets/57421823/bdfbc044-035e-4a6f-95d6-72ee30edb419)

Browser's console:

![image](https://github.com/kamranasif506/Book-A-Doctor/assets/57421823/0a9f0678-96d2-41b8-958e-f8d401dab6cd)

Pry output when inserting a breakpoint in the "create" method in the backend:

![image](https://github.com/kamranasif506/Book-A-Doctor/assets/57421823/5a071745-403d-4423-adf8-49a614204263)

For some reason, the user is coming as nil.  


These are the created users:

![image](https://github.com/kamranasif506/Book-A-Doctor/assets/57421823/69b7acae-5f5b-4399-8e85-4803d7e76b2a)

If I try to reserve with one of the original users created by the seeds.rb file (hassan@gmail.com id:4):

on login:

![image](https://github.com/kamranasif506/Book-A-Doctor/assets/57421823/510ae24c-099a-45cc-96ae-e6b6a3597121)


trying to reserve:

![image](https://github.com/kamranasif506/Book-A-Doctor/assets/57421823/2dfa8f8b-5595-48c5-9502-ba7361977c80)

![image](https://github.com/kamranasif506/Book-A-Doctor/assets/57421823/d58b6aa1-89c9-4e67-9ec9-f26455e7b4cd)

I tried with multiple browsers, with 2 computers.  I also tried multiple doctors and locations combinations, with multiple users.

Reply:
# UPDATE: I managed to reserve a doctor

1. After logging in, I went to this address in my browser:  localhost:3000/current_user

2. After the last step, I tried to make a reservation and now Rails is successfully acknowledging the current_user:

![image](https://github.com/kamranasif506/Book-A-Doctor/assets/57421823/f2c81fc6-5c49-4668-8c9d-44affaa7dfd0)

![image](https://github.com/kamranasif506/Book-A-Doctor/assets/57421823/e67429e4-0439-4920-a2ce-96dd757d72d5)

3. ... And now the reservation works:

![image](https://github.com/kamranasif506/Book-A-Doctor/assets/57421823/4f1cdcda-cfe8-4ae2-bd74-0e7d25aad7cc)

![image](https://github.com/kamranasif506/Book-A-Doctor/assets/57421823/70f8fb55-b004-4d31-81a9-b09acb5347f3)

4. As long as I don't change the user, I can reserve all the doctors I want:

![image](https://github.com/kamranasif506/Book-A-Doctor/assets/57421823/5f32d008-28d8-4e3e-a267-a6dfd623c0aa)

5. If I login with another user, the problem starts again:

![image](https://github.com/kamranasif506/Book-A-Doctor/assets/57421823/9d6f5b2c-9b65-4c45-a9b7-272c1c2c1c57)

6. The problem persists again until I go to localhost:3000/current_user:


![image](https://github.com/kamranasif506/Book-A-Doctor/assets/57421823/2300f945-6e55-4866-a66e-c65f8a074d1f)

Then I try again and it works!

![image](https://github.com/kamranasif506/Book-A-Doctor/assets/57421823/448f547f-4bfc-49b4-8b67-970e9740c71a)

Reply:
I have resolved this issue now .. and request for another review,
the issue was with the access token. it was not sending properly with the request which causes issues

Reply:
Therefore, it seems I was able to bypass the authorization 😁